### PR TITLE
Added servlets to process new signups and logins

### PR DESCRIPTION
**Pull request by keyuyan1145**
I added two servlets to add the new user info into the database for signup and verify the login credentials. I also added the GET request to pull the user info for the profile page, but this might need to be modified later to better interact with the front-end. So far, the password is not encrypted. If time permits, this could be added at a later time. Once the user has logged in, the username and the unique id of the user are stored in a cookie.

### CODE REVIEW

File: src/main/java/com/google/sps/java/User.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  private final String number;
+  private final String email;
+
+
+  public User(String username, String name, String email, String number, 
+      String organization) {
+    this.username = username;
+    this.name = name;
+    this.email = email;
+    this.number = number;
```
Comment (sofifch) [inline src/main/java/com/google/sps/java/User.java:17]: Is number referring to phone number? or some kind of id for the user?

File: src/main/java/com/google/sps/servlets/NewUserServlet.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.gson.Gson;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.*;
```
Comment (sofifch) [inline src/main/java/com/google/sps/servlets/NewUserServlet.java]: I think you should avoid using * and specify what you are importing

File: src/main/java/com/google/sps/servlets/NewUserServlet.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    userEntity.setProperty("username", username);
+    userEntity.setProperty("email", email);
+    userEntity.setProperty("password", password);
+    userEntity.setProperty("contactNumber", contactNum);
+    userEntity.setProperty("organization", organization);    
+    
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(userEntity);
+
+    System.out.println("successfully added NEW USER into database");
```
Comment (sofifch) [inline src/main/java/com/google/sps/servlets/NewUserServlet.java]: I know this is for you to check functionality, but let's not include it when merging with master, or if you want, log it into the js console.

File: src/main/java/com/google/sps/servlets/UserServlet.java [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.gson.Gson;
+import com.google.sps.team36.User;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.Cookie;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.*;
```
Comment (sofifch) [inline src/main/java/com/google/sps/servlets/UserServlet.java]: Same as above!

File: src/main/java/com/google/sps/servlets/UserServlet.java [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+    String password = request.getParameter("password");
+
+    Query query = new Query("user")
+      .setFilter(
+          CompositeFilterOperator.and(FilterOperator.EQUAL.of("username", username), FilterOperator.EQUAL.of("password", password)));
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery result = datastore.prepare(query);
+
+    System.out.println("Result: "+result);
```
Comment (sofifch) [inline src/main/java/com/google/sps/servlets/UserServlet.java:51]: Same as above, probably console log it.

File: src/main/java/com/google/sps/java/User.java (hunk 1 above)
Comment (keyuyan1145) [reply src/main/java/com/google/sps/java/User.java:17]: Number refers to phone number

File: src/main/java/com/google/sps/servlets/NewUserServlet.java (hunk 2 above)
Comment (keyuyan1145) [reply src/main/java/com/google/sps/servlets/NewUserServlet.java]: I will remove this import, didn't use it in the code.

File: src/main/java/com/google/sps/servlets/NewUserServlet.java (hunk 3 above)
Comment (keyuyan1145) [reply src/main/java/com/google/sps/servlets/NewUserServlet.java]: I'll remove it.

**Review by sofifch — APPROVED** [review]
Let's merge this one, and then mine, so that we can get on the filtering, messaging and profile view!

**Pull request merged**